src/integrationTest/java/uk/gov/hmcts/reform/bulkscan/payment/processor/client/payhub/PayHubClientPutPaymentTest.java

```diff
@@ -0,0 +1,125 @@
+package uk.gov.hmcts.reform.bulkscan.payment.processor.client.payhub;
+
+import com.github.tomakehurst.wiremock.client.ResponseDefinitionBuilder;
+import com.google.common.collect.ImmutableList;
+import org.assertj.core.api.SoftAssertions;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import uk.gov.hmcts.reform.bulkscan.payment.processor.client.payhub.request.CaseReferenceRequest;
+import uk.gov.hmcts.reform.bulkscan.payment.processor.config.IntegrationTest;
+
+import java.util.List;
+import java.util.function.Function;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.aResponse;
+import static com.github.tomakehurst.wiremock.client.WireMock.equalTo;
+import static com.github.tomakehurst.wiremock.client.WireMock.equalToJson;
+import static com.github.tomakehurst.wiremock.client.WireMock.okJson;
+import static com.github.tomakehurst.wiremock.client.WireMock.put;
+import static com.github.tomakehurst.wiremock.client.WireMock.stubFor;
+import static com.github.tomakehurst.wiremock.client.WireMock.urlEqualTo;
+import static java.util.UUID.randomUUID;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.catchThrowable;
+import static org.springframework.http.HttpStatus.BAD_REQUEST;
+import static org.springframework.http.HttpStatus.INTERNAL_SERVER_ERROR;
+
+@IntegrationTest
+public class PayHubClientPutPaymentTest {
+
+    private static final String CASE_REF_REQUEST_JSON = "{ \"ccd_case_number\": \"12321321\"}";
+
+    @Autowired
+    private PayHubClient client;
+
+    @Test
+    public void should_return_ok_when_everything_is_ok_ignore_response_body() {
+        // given
+        String s2sToken = randomUUID().toString();
+
+        stubWithRequestAndResponse(
+            s2sToken,
+            "98765342",
+            CASE_REF_REQUEST_JSON,
+            okJson("{\"a\":\"1\"}")
+        );
+
+        // when
+        ResponseEntity updateResponse =
+            client.putPayments(
+                s2sToken,
+                "98765342",
+                new CaseReferenceRequest("12321321")
+            );
+
+        // then
+        assertThat(updateResponse.getStatusCodeValue()).isEqualTo(200);
+
+        //Ignore response body for now !!!
+        assertThat(updateResponse.getBody()).isNull();
+    }
+
+    @Test
+    public void should_return_PayHubClientException_for_errors() {
+        List<HttpStatus> tests = ImmutableList.of(BAD_REQUEST, BAD_REQUEST, INTERNAL_SERVER_ERROR);
+        SoftAssertions softly = new SoftAssertions();
+        tests.stream().forEach(t -> testError(softly, t));
+        softly.assertAll();
+    }
```

Comment:
We could just use parameterised test with enum source ;). This way we can just squash that private method into one test case

Reply:
ParameterizedTest used